runtime/commonMain/src/kotlinx/serialization/internal/CollectionSerializers.kt

```diff
@@ -140,6 +140,45 @@ sealed class MapLikeSerializer<TKey, TVal, TCollection, TBuilder: MutableMap<TKe
     }
 }
 
+abstract class PrimitiveArraySerializer<TElement, TCollection, TBuilder : PrimitiveArraySerializer<TElement, TCollection, TBuilder>.Builder>
+internal constructor(
+    primitiveSerializer: KSerializer<TElement>,
+    primitiveDescriptor: PrimitiveDescriptor
+) : ListLikeSerializer<TElement, TCollection, TBuilder>(primitiveSerializer) {
+    final override val descriptor = PrimitiveArrayDescriptor(primitiveDescriptor)
+
+    abstract inner class Builder internal constructor() {
+        abstract val pos: Int
```

Comment:
Please do not use shorthands for overridable properties (same for `objSize` and `objIterator`)